proposals/p1885.md

```diff
@@ -744,73 +738,52 @@ class IterateIterator {
       : container_(other.container_), cursor_(other.cursor_),
       value_(other) {}
 
-  auto operator*() const -> const typename Iterable::ElementType& {
+  auto operator*() const -> const typename Iterate::ElementType& {
     assert(cursor_);
     return value_;
   }
-  auto operator++() -> IterateIterator<Iterable>& {
+  auto operator++() -> IterateIterator<Iterate>& {
     next();
     return *this;
   }
-  // `true` if both are equal or invalid/empty
-  auto operator==(const IterateIterator<Iterable>& rhs) -> bool {
+  auto operator==(const IterateIterator<Iterate>& rhs) -> bool {
     return container_ == rhs.container_ && cursor_ == rhs.cursor_;
   }
-  auto operator!=(const IterateIterator<Iterable>& rhs) -> bool {
+  auto operator!=(const IterateIterator<Iterate>& rhs) -> bool {
     return !(*this == rhs);
   }
 
  private:
   void next() {
     assert(cursor_);
-    if (const auto val = container_->Next(*cursor_); val) {
-      value_ = *val;
+    if (const auto v = container_->Next(*cursor_); v) {
+      value_ = *v;
```

Comment:
```suggestion
      value_ = *std::move(v);
```

This will enable the code to work with move-only types.

Reply:
Good point, thanks.